src/utils/converters.ts

```diff
@@ -2,20 +2,40 @@ import BigNumber from 'bignumber.js';
 
 const MASSA_SCALING_FACTOR = 9;
 
-export const toMAS = (inputAmount: BigNumber | number | string): number => {
-  const amount = new BigNumber(inputAmount);
+/**
+ * Convert any value expressed in Massa back to nanoMassa
+ * @param amountInMassa any amount in Massa
+ * @returns bigint representing the amount expressed in nanoMassa
+ */
+export const fromMAS = (
+  amountInMassa: number | string | BigNumber | bigint,
+): bigint => {
+  const amount = new BigNumber(amountInMassa.toString());
   const scaleFactor = new BigNumber(10).pow(
     new BigNumber(MASSA_SCALING_FACTOR),
   );
   const amountScaled = amount.times(scaleFactor);
-  return parseInt(amountScaled.toFixed(0));
+  return BigInt(amountScaled.toFixed(0));
 };
 
-export const fromMAS = (inputAmount: number | string): BigNumber => {
-  const amount = new BigNumber(inputAmount);
+/**
+ * Convert any value expressed in nano Massa back to Massa
+ * @param amountInNanoMassa any amount in nanoMassa
+ * @returns BigNumber representing the amount expressed in Massa
+ */
+export const toMAS = (
+  amountInNanoMassa: number | string | BigNumber | bigint,
+): BigNumber => {
```

Comment:
why toMAS returns BigNumber whereas fromMAS returns bigint?

Reply:
because e.g. 300_000_000  is 0.3 massa. and 0.3 can go into a BigNumber.